File: Client/src/app/shared/models/user.model.ts
import './role.model'
import { IRole } from './role.model';
export interface IUser {
    userName: string;
    normalizedUserName: string;
    email: string;
    name: string;
    surname: string;
    accessFailedCount: number;
    isLockoutEnabled: boolean;
    lockoutEndDateUtc: Date;
    normalizedEmail: string;
    passwordHash: string;
    securityStamp: string;
    emailConfirmed: boolean;
    roles: IRole[];
    isActive: boolean;
    id: number
}

export interface IUserEdit {
    id?: number;
    name: string;
    surname: string;
    username: string;
    emailAddress: string;
    password: string;
    isActive: boolean;
    shouldChangePasswordOnNextLogin: boolean;
}
export interface IUserRole {
    roleId: number;
    roleName: string;
    roleDisplayName: string;
    isAssigned: boolean;
}

export interface IUserForCreateOrEdit {
    assignedRoleCount: number;
    isEditMode: boolean;
    user: IUserEdit;
    roles: IUserRole[];
}

export class CreateOrUpdateUser {
    randomPassword:boolean = true;
    user: IUserEdit;
    assignedRoleNames: string[];
    sendActivationEmail: boolean;
}

export interface IUserProfile{
    id: number;
    name: string;
    surname: string;
    username: string;
    email: string;
}

export class UserProfile implements IUserProfile{
    constructor(public id:number = 0, public name:string = '', public surname: string = '', public username:string = '', public email:string=''){

    }
}

File: ApiServer/Model/views/UserModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ApiServer.Model.views
{
    public static class UserModel
    {
        public class User
        {
            public long Id { get; set; }
            public string UserName { get; set; }
            public string NormalizedUserName { get; set; }
            public string Email { get; set; }
            public string Name { get; set; }
            public string Surname { get; set; }
            public int? AccessFailedCount { get; set; }
            public bool? IsLockoutEnabled { get; set; }
            public DateTime? LockoutEndDateUtc { get; set; }
            public bool IsActive { get; set; }
            public string NormalizedEmail { get; set; }
            public string PasswordHash { get; set; }
            public string SecurityStamp { get; set; }
            public bool? EmailConfirmed { get; set; }
            public RoleModel.RoleBase Roles { get; set; }
        }

        public class UserForCreateOrEdit
        {
            public int AssignedRoleCount { get; set; }
            public bool isEditMode { get; set; }
            public UserEdit User { get; set; }
            public List<UserRole> Roles { get; set; }
        }

        public class CreateOrUpdateUser
        {
            public bool RandomPassword { get; set; }
            public bool SendActivationEmail { get; set; }
            public UserEdit User { get; set; }
            public List<string> AssignedRoleNames { get; set; }
            public List<string> UnAssignedRoleNames { get; set; }
        }

        public class UserEdit
        {
           public long Id { get; set; }
           public string Name { get; set; }
           public string Surname { get; set; }
           public string Username { get; set; }
            public string EmailAddress { get; set; }
            public string Password { get; set; }
            public bool IsActive { get; set; }
            public bool ShouldChangePasswordOnNextLogin { get; set; }
        }

        public class UserRole
        {
            public int RoleId { get; set; }
            public string RoleName { get; set; }
            public string RoleDisplayName { get; set; }
            public bool IsAssigned { get; set; }
        }
    }

}
